pkg/prometheus/promcfg.go

```diff
@@ -216,12 +219,15 @@ func generateServiceMonitorConfig(version semver.Version, m *v1.ServiceMonitor,
 	switch version.Major {
 	case 1:
 		if version.Minor < 7 {
-			cfg = append(cfg, k8sSDAllNamespaces())
+			if apiserverConfig != nil {
+				log.Printf("custom apiserver config is set but it will not take effect if prometheus version is < 1.7")
```

Comment:
We are not using the standard logger anymore, but instead use [leveled logging](https://github.com/coreos/prometheus-operator/blob/master/pkg/prometheus/operator.go#L61).

For example: https://github.com/coreos/prometheus-operator/blob/master/pkg/prometheus/operator.go#L731-L733

Reply:
Thank you for reply. But I found that functions in `prom_cfg.go` are all "static". They all have no logger object like the one in operator object. I can't simply add a line for log it. I wonder how to use this leveled logging package without create a local logger object nor change these static functions to operator's method. What do you think?

Reply:
We should bind these methods to a configGenerator struct that is instantiated in the reconcile loop and that gets the logger passed. Then the bound methods can just use the logger from the struct they’re bound to.

Reply:
OK, Shuold I do this in current PR, or create another one?

Reply:
Done. cc @brancz